src/index.tsx

```diff
@@ -497,10 +497,32 @@ export class Rnd extends React.Component<Props, State> {
     const { left, top } = this.getOffsetFromParent();
     let draggablePosition;
     if (position) {
```

Comment:
@chathuraa Sorry, could you please separate these block to another function or method like `createNewPosition` 🙏

Reply:
Hey @bokuweb ,

I had to update some tests, I couldnt figure out the reason for '8px' discrepancy. I managed to get pretty much everything working. apart from the uncontrolled bounds. 

Thanks,
Chat.

Reply:
Ahhh..Is it due to default margin? If you set `html, body { padding: 0,  margin: 0 };` to `./fixture.url`, will the results be as you expected?

Reply:
Hey @bokuweb,

oh okay, that makes sense. I've updated the getOffsetFromParent  and  the tests so the padding and margins become irrelevant. 

The update on the getOffsetFromParent, seems to cause an issue with uncontrolled bounds. I cant seems to  find where the issue is. Any suggestions?

Thanks,
Chat.

Reply:
I tried that. however the issue was there on almost every where the getOffsetFromParent been used. 
ie: 
componentDidMount,
dragStart,
onDrag,
onDragStop,
etc

So decided to update the function to return left/top based on axis restrictions, rather than setting the logic on each function.

Re: the bounds, it only affecting the uncontrolled bounds. all the other scenarios including controlled bounds are working.

Reply:
Hmmmm, I'll try to fix it 🙏 So, this issue is complex....